code/modules/projectiles/projectile/beams.dm

```diff
@@ -184,3 +184,232 @@
 		var/mob/living/carbon/M = target
 		M.visible_message("<span class='danger'>[M] explodes into a shower of gibs!</span>")
 		M.gib()
+
+/obj/item/projectile/beam/heavy
+	name = "\improper heavy laser beam"
+	icon_state = "heavylaser"
+	damage = 30
+	range = 15
+	irradiate = 30
+	armour_penetration = 20
+	dismemberment = 10
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/green_laser
+	light_color = LIGHT_COLOR_RED
+	tracer_type = /obj/effect/projectile/tracer/xray
+	muzzle_type = /obj/effect/projectile/muzzle/xray
+	impact_type = /obj/effect/projectile/impact/xray
+	pass_flags = PASSTABLE
+
+/obj/item/projectile/beam/heavy/on_hit(atom/target, blocked = FALSE)
+	. = BULLET_ACT_FORCE_PIERCE
+	..()
+
+/obj/item/projectile/beam/shock
+	name = "\improper charged beam"
+	icon_state = "spark"
+	damage = 22
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/green_laser
+	light_color = LIGHT_COLOR_CYAN
+	tracer_type = /obj/effect/projectile/tracer/xray
+	muzzle_type = /obj/effect/projectile/muzzle/xray
+	impact_type = /obj/effect/projectile/impact/xray
+
+/obj/item/projectile/beam/shock/on_hit(atom/target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.electrocute_act(15, C, 1, FALSE, FALSE, FALSE, FALSE, FALSE)
+		if(prob(20))
+			C.dropItemToGround(C.get_active_held_item())
+		C.confused += 6
+
+/obj/item/projectile/beam/bitcoin
+	name = "\improper bitcoin stealing beam"
+	icon_state = "bitcoin"
+	damage = 0
+	nodamage = TRUE
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/green_laser
+	light_color = LIGHT_COLOR_GREEN
+	tracer_type = /obj/effect/projectile/tracer/xray
+	muzzle_type = /obj/effect/projectile/muzzle/xray
+	impact_type = /obj/effect/projectile/impact/xray
+
+/obj/item/projectile/beam/bitcoin/on_hit(atom/target)
+	. = ..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		var/obj/item/card/id/C = H.get_idcard(TRUE)
+		if(C && C.registered_account)
+			C.registered_account.adjust_money(-100)
+		else
+			H.adjustStaminaLoss(40)
+		if(ishuman(firer))
+			var/mob/living/carbon/human/FH = firer
+			var/obj/item/card/id/FC = FH.get_idcard(TRUE)
+			if(FC && FC.registered_account)
+				FC.registered_account.adjust_money(100)

```

Comment:
shouldn't this only add however much money you stole to your account?

also, does this even check if the target's account has any money in it?